CA-223651 fix meta data for VM and SR owner

lindig opened the pull request:
This PR fixes how meta data is created for a VM and VR data source owner and changes the integration test to test the VM case (was HOST). Once this goes in, I'll tag this and update the SPEC file.

lindig committed:
CA-223651 fix JSON meta data for owner

This commit fixes the owner meta data for VMs and SRs. The string
representing them need to include the UUID. The testing code rrdclient.c
was changed to now use RRD_VM in order to test this as part of the
integration test.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

lindig committed:
CA-223651 make integration test more reliable

Wait 2 seconds to give the client a chance to write data before trying
to read it.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

robertbreker commented on librrd.c:
```diff
@@ -134,16 +134,17 @@ json_for_source(RRD_SOURCE * source)
         break;
     case RRD_VM:
-        owner = "vm";
+        snprintf(owner, sizeof owner, "vm %s", source->owner_uuid);
```
I think to prevent buffer overruns and make coverity happy, we should consider the following:

```
owner[128] = {0};
...
snprintf(owner, sizeof(owner)-1, "vm %s", source->owner_uuid))
```

However, we'd only ever run into this, if the library user provides a too long uuid.

lindig replied:
`snprintf(owner,n,fmt)` does not overwrite this buffer. It writes at most `n-1` characters plus the null byte. 

http://en.cppreference.com/w/c/io/fprintf

robertbreker commented:
LGTM despite the minor comment.

lindig committed:
CA-223651 initialise local buffer owner[]

Use a static initialiser for owner[] buffer. This is not strictly
necessary as snprintf() is guaranteed to terminate it with a null byte.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

robertbreker commented:
LGTM
